src/modules/fw_path_navigation/FixedwingPositionControl.cpp

```diff
@@ -709,8 +709,24 @@ FixedwingPositionControl::set_control_mode_current(const hrt_abstime &now)
 
 	_skipping_takeoff_detection = false;
 
-	if (((_control_mode.flag_control_auto_enabled && _control_mode.flag_control_position_enabled) ||
-	     _control_mode.flag_control_offboard_enabled) && _position_setpoint_current_valid) {
+	if (_control_mode.flag_control_offboard_enabled && _position_setpoint_current_valid) {
+		if (_control_mode.flag_control_position_enabled) {
```

Comment:
What do you expect to happen when this if(_control_mode.flag_control_position_enabled) is false?